packages/react-app/src/models/RoleCache.ts

```diff
@@ -0,0 +1,19 @@
+import { Role } from '@app/types/Role';
+import Mongoose, { Model } from 'mongoose';
+
+export interface IRoleCache{
+  sessionHash: string;
+  createdAt: number;
+  tte: number;
+  roles: Role[]
+}
+
+export const RoleSchema = new Mongoose.Schema({
+	sessionHash: String,
+	createdAt: Number,
+	tte: Number,
```

Comment:
If your check to see whether the cache is still valid is:
```
return cache.createdAt + cache.tte <= new Date().getTime();
```

is there a reason you went with `createdAt` and `tte` as fields, instead of a singular field `expiresAt` ?